src/extension.ts

```diff
@@ -1,53 +1,130 @@
-	// The commandId parameter must match the command field in package.json
-	const disposable1 = vscode.commands.registerCommand('vscode-manim.helloData', () => {
-		// The code you place here will be executed every time your command is executed
-		// Display a message box to the user
-        // terminal.show();
-		vscode.window.showInformationMessage('Hello Data from vscode-manim!');
-	});
-
-	const disposable2 = vscode.commands.registerCommand('vscode-manim.checkpointPaste', async () => {
-		// The code you place here will be executed every time your command is executed
-		// Display a message box to the user
-		const terminal = vscode.window.activeTerminal || vscode.window.createTerminal();
-		const editor = vscode.window.activeTextEditor;
-		if (editor) {
-			// Get the selected text
-			const selection = editor.selection;
-			const selectedText = editor.document.getText(selection);
+	/**
+	 * Command to preview the Manim code of the cell where the cursor is placed
+	 * (when accessed via the command pallette) or the code of the cell where
+	 * the codelens was clicked.
+	 */
+	const previewManimCell = vscode.commands.registerCommand('vscode-manim.previewManimCell',
+		(cellCode: string | undefined) => {
+			// User has executed the command via command pallette
+			if (cellCode === undefined) {
+				const editor = vscode.window.activeTextEditor;
+				if (!editor) {
+					vscode.window.showErrorMessage('No opened file found. Place your cursor in a Manim cell.');
+					return;
+				}
+				const document = editor.document;
+				
+				// Get the code of the cell where the cursor is placed
+				const cursorLine = editor.selection.active.line;
+				const range = ManimCellRanges.getCellRangeAtLine(document, cursorLine);
+				if (!range) {
+					vscode.window.showErrorMessage('Place your cursor in a Manim cell.');
+					return;
+				}
+				cellCode = document.getText(range);
+			}
 
-			// Copy the selected text to the clipboard
-			await vscode.env.clipboard.writeText(selectedText);
+			previewCode(cellCode);
+		});
 	
-			// Create or show the terminal
-			const terminal = vscode.window.activeTerminal || vscode.window.createTerminal();
-			
-			// Send the checkpoint_paste() command
-			// terminal.sendText('checkpoint_paste()', false);
-			terminal.sendText('checkpoint_paste()');
-	
-			vscode.window.showInformationMessage('Copied selected code and sent checkpoint_paste() to manim terminal');
-		} else {
-			vscode.window.showErrorMessage('No text is selected');
+	// TODO: Why do we even need this "preview selection feature"? I think that
+	// we rather want a "Preview active Manim Cell up to cursor position" feature.
+	// Otherwise, we need to remind the user that they have to include the start
+	// of the cell in the selection, which is pretty annoying to do manually.
+	//
+	// I rather want to place my cursor anywhere in a cell and press a key
+	// to run the cell up to the cursor position.
```

Comment:
I'm questioning the "preview cell" feature here. What do you think about the proposal?

Reply:
Meanwhile I've removed this TODO note from the code. The comment refers to the `previewSelection()` method of `extension.ts`.

Reply:
opened a new issue #15 for that ;)